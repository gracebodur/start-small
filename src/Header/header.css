.App-header {
    background-color: #68a1e0;
    /* min-height: 10vh; */
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: 70px;
    color: #ffd55d;
    font-family: 'Lilita One', cursive;
    letter-spacing: 3px;
  }
  
  h3 {
    /* display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center; */
    font-size: 30px;
    color: white;
    font-family: 'Rosario', sans-serif;
  }