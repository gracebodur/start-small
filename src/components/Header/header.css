.App-header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: 50px;
    letter-spacing: 3px;
  }

  .nav-link {
    text-decoration: none;
  }

  .Nav {
    display: flex;
    justify-content: flex-end;
    font-family: var(--ff-seconday);
    color: var(--clr-purple);
    padding: 1rem;
    font-size: 1.25rem;
}

p {
    padding-right: 30px;
}

.dim {
    opacity: 1;
    transition: opacity .15s ease-in;
}

.dim:hover,
.dim:focus {
    opacity: .5;
    transition: opacity .15s ease-in;
}

.dim:active {
  opacity: .7; transition: opacity .15s ease-out;
}




































