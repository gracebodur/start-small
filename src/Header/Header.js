import React from 'react'
import './header.css'

function Header() {
    return (
    <div>
    <header className="App-header">
        <h1>START small</h1>
        <h3>A learning platform that aims to helps improve every child's educationÂ </h3>
    </header>
    </div>
    )
}

export default Header