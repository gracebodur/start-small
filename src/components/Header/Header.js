import React, {Component} from 'react'
import TokenService from '../../services/token-service'
import {Link } from 'react-router-dom'
import Logo from '../Logo/Logo'
import './Header.css'

class Header extends Component {

    state = {isLoggedIn: false}

    handleLogOutClick = () => {
        TokenService.clearAuthToken()
    }

    renderLogoutLink() {
        return(
            <div className='Header__logged-in'>
                <Link 
                    onClick={this.handleLogOutClick} 
                    to='/'>
                    Log out
                </Link>
            </div>
        )
    }

    renderLoginLink() {
        return(
            <div className='Header__not-logged-in'>
                <Link to='/login'>
                    Log in
                </Link>

                <Link to='/register'>
                    Register
                </Link> 
            </div>

        )
    }

    render() {
        return (
            <nav className='Header'>
                <Logo />
                { TokenService.hasAuthToken()
                ? this.renderLogoutLink()
                : this.renderLoginLink() }
            </nav>
        )
    }
}


export default Header

