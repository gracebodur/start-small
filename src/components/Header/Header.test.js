import React from "react";
import toJson from "enzyme-to-json";
import Header from "./Header";

describe(`Header component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a Header by default", () => {
    const wrapper = <Header />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Header given props", () => {
    const wrapper = <Header {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
