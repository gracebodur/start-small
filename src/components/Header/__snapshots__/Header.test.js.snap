// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `undefined`;

exports[`Header component renders the Header given props 1`] = `undefined`;
